cli/http_util.rs

```diff
@@ -84,16 +132,23 @@ pub enum FetchOnceResult {
 pub async fn fetch_once(
   client: Client,
   url: &Url,
-  cached_etag: Option<String>,
+  maybe_etag: Option<String>,
+  maybe_auth_token: Option<String>,
```

Comment:
Maybe not for now, but this method is getting lots of args, we might want to put them into a struct.

Why doesn't the `maybe_auth_token` argument not take `Option<AuthToken>`?

Reply:
Because it isn't the structure that holds all the auth tokens.  I can create a type alias of `AuthToken` for `String` but it seems a bit trivial.

Reply:
Actually I see what I've done... ok... let me modify it so we pass `AuthToken` and set the display or something to generate the header value.

Reply:
I have also moved to a struct for the args, otherwise we would likely never do it.